src/waypoint.jsx

```diff
@@ -34,11 +34,24 @@ function getCurrentPosition(bounds) {
     return POSITIONS.invisible;
   }
 
+  // top is within the screen
   if (bounds.viewportTop <= bounds.waypointTop &&
       bounds.waypointTop <= bounds.viewportBottom) {
     return POSITIONS.inside;
   }
 
+  // bottom is within the screen
+  if (bounds.viewportTop <= bounds.waypointBottom &&
+      bounds.waypointBottom <= bounds.viewportBottom) {
+    return POSITIONS.inside;
+  }
+
+  // top is above the screen and bottom is below the screen
```

Comment:
I think this issue goes beyond this PR, but we should be consistent about the name that we give the "containing element that is scrollable." Here it's being called "screen," the prop to manually set it it is "scrollableAncestor," and when the offsets are taken into account, its called the "viewport." Does anyone else think this could affect someone's understanding of the code? If not, then maybe it's no big deal.